agent/xds/listeners.go

```diff
@@ -254,6 +257,12 @@ func (s *Server) listenersFromSnapshotGateway(cfgSnap *proxycfg.ConfigSnapshot,
 			if err != nil {
 				return nil, err
 			}
+		case structs.ServiceKindIngressGateway:
+			listeners, err := s.listenersFromSnapshotIngressGateway(a.Address, cfgSnap)
```

Comment:
Can this be renamed to match the other gateways? `s.makeIngressGatewayListeners`